PostHog/Autocapture/AutocaptureEventProcessing.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  AutocaptureEventProcessing.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 22/10/2024.
+//
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    import UIKit
+
+    protocol AutocaptureEventProcessing: AnyObject {
+        func process(source: PostHogAutocaptureIntegration.EventData.EventSource, event: PostHogAutocaptureIntegration.EventData)
+    }
+
+    class PostHogAutocaptureEventProcessor: AutocaptureEventProcessing {
+        private weak var postHogInstance: PostHogSDK?
```

Comment:
can we just use the `PostHogSDK.shared` directly? I know it's not great for testability, but we don't need to handle nullable things, I have an idea to improve this.

Reply:
All valid points, I'd focus on getting it working with the shared instance for now, but indeed the multiple instances is an issue (it is already since it does not work).
I'd say that all "integrations/autocapture" etc work with the shared instance and if you create more instances, you'd need to do everything (capture events, etc) all by hand since it's an edge case and most likely is used by SDKs/Libs, which should not capture events from the hosting app anyway.
Yes, you can only swizzle once, or, more times but the last swizzle wins and that leads to more complexity to get everything working, so I'd rather not do this unless we have to at some point.
The auto capture can be an integration, and the config has a list of integrations, for example you can see how this works on Android
https://github.com/PostHog/posthog-android/blob/351a86b7c7caf9c72aad684d3721fad2872cfa2c/posthog/src/main/java/com/posthog/PostHogConfig.kt#L156-L169